packages/build/src/lib/schema-parser/middleware/checkCache.ts

```diff
@@ -0,0 +1,56 @@
+import { ConfigurationError } from '@vulcan-sql/core';
+import { RawAPISchema, SchemaParserMiddleware } from './middleware';
+import { CACHE_METADATA_NAME } from './constants';
+import ms, { StringValue } from 'ms';
+
+export class CheckCache extends SchemaParserMiddleware {
+  public async handle(schemas: RawAPISchema, next: () => Promise<void>) {
+    // check .yaml file has cache configuration
+    const caches = schemas?.cache;
+    const metadata = schemas.metadata;
+    const isUsedCache = metadata?.[CACHE_METADATA_NAME]?.['isUsedTag'];
+    if (!isUsedCache && !caches) return next();
+
+    // throw if cache not used in .sql file
+    if (isUsedCache && !caches) {
+      throw new ConfigurationError(
+        "Cache feature was used in SQL file, but didn't find the cache configurations in YAML file."
+      );
+    }
+
+    if (caches) {
+      caches.forEach(({ refreshTime, refreshExpression }): void => {
+        // refreshTime and refreshExpression can not be set at the same time
+        if (refreshTime && refreshExpression) {
+          throw new ConfigurationError(
+            'can not configure refreshTime and refreshExpression at the same time, please pick one'
+          );
+        }
+        // the "every" in refreshTime or refreshExpression should be valid
+        const timeInterval =
+          refreshTime?.['every'] || refreshExpression?.['every'];
+        if (timeInterval) {
+          let interval: number;
+          try {
+            interval = ms(timeInterval as StringValue);
+          } catch (error) {
+            throw new ConfigurationError(
+              'invalid refreshTime representation, check node library "ms" for the valid representation'
```

Comment:
Here the value of `every` may from `refreshTime` and `refreshExpression`, but according to your message,m why you could be sure it from `refreshTime` ? 🤔 

I may suggest you check `refreshTime` and `refreshExpression`by different conditions and throw a different error messages, or the user will be confused.

Reply:
Make sense.

Well... first I think we can assume that we will have a document to introduce the Cache layer feature, which will contain a .yaml file of how to set the cache and describe the relevant setting parameters.

I suggest...can reference the error message format:
E.g.,
 - `The "every" of the "refreshExpression" is invalid.`
 - `Invalid "every" time format of the "refreshTime": ${error?.message}`

- - - -
Other ideas - If can provide a suitable refresh time to the user. 😂

Reply:
@fredalai  I think both are great suggestions 👍 

Because the second part is throwing the `ms` original error message, we may need to check what the original message is and message content is suitable or not to use the second suggestion, cc @onlyjackfrost

Reply:
@fredalai  You mean to provide some options ["5m", "15m", "30m", "1h",...] and not allow users to define them?

@kokokuo 
I think we can hide the "ms" library in message if we have a document to introduce the Cache layer, describe the usage there, and give some examples.

Reply:
HI @onlyjackfrost -

No, we want users to define the refresh time according to their own needs.

I mean...we can provide a suitable refresh time, and add it to the message if not specify every refresh time to suggest a user.
But I asked Eason, and got the previously discussed content is we will not provide a default value to a user.